【バグ】 player/process-death.cpp compiler error with clang and English

## 環境

- OS: macOS 11.6.2
- 3.0.0Alpha54 and changes up to and including c786fdc124d52f5fe719d11c2a09dc0716c2d25d + Mac-specific changes from https://github.com/backwardsEric/hengband

## バグの詳細

With clang 13.0.0 (from Xcode 13.2.1) and compiling for English, there's a compilation error when compiling src/player/process-death.cpp:

player/process-death.cpp:192:47: error: cannot pass object of non-trivial type 'std::string' (aka 'basic_string<char>') through variadic function; call will abort at runtime [-Wnon-pod-varargs]

## 再現手順

バグを発生させる手順が分かっていれば記載

In the top-level directory run the following:

1. ./bootstrap
2. ./configure --enable-cocoa --disable-japanese
3. make install

## 期待される正常な動作

Compilation works for English as it does for Japanese.

--- a/src/player/process-death.cpp
+++ b/src/player/process-death.cpp
@@ -189,7 +189,7 @@ static void show_tomb_detail(PlayerType *player_ptr, char *buf)
     center_string(buf, tomb_message);
     put_str(buf, 14, 11);
 
-    shape_buffer(format("by %s.", player_ptr->died_from), GRAVE_LINE_WIDTH + 1, tomb_message, sizeof(tomb_message));
+    shape_buffer(format("by %s.", player_ptr->died_from.c_str()), GRAVE_LINE_WIDTH + 1, tomb_message, sizeof(tomb_message));
     center_string(buf, tomb_message);
     char *t;
     put_str(buf, 15, 11);

